samples/weather_function/src/main/kotlin/org/example/weather_func/main.kt

```diff
@@ -0,0 +1,123 @@
+package org.example.weather_func
+
+import platform.posix.*
+import kotlinx.cinterop.allocArray
+import kotlinx.cinterop.ByteVar
+import kotlinx.cinterop.memScoped
+import kotlinx.cinterop.toKString
+import kotlin.system.exitProcess
+
+private val API_KEY by lazy { fetchApiKey() }
+
+fun main(args: Array<String>) {
+	val input = readLine()
+	val location = fetchLocationArg(input)
+	val jsonFile = fetchJsonFileArg(input)
+
+	if (location.isNotEmpty()) {
+		printFromWeatherService(location)
+	} else if (jsonFile.isNotEmpty()) {
+		checkFile(jsonFile)
+		printJsonFile(jsonFile)
+	} else {
+		handleEmptyInput()
+	}
+	println("Exiting...")
+}
+
+private fun handleEmptyInput() {
+	println(
+		"""
+		| Weather - A OpenFaaS Serverless Function that outputs weather information.
+		| Usage examples (pass one of the following as input to the function):
+		|   * Location (uses the Open Weather Map service), eg:
+		|       -l="christchurch,nz"
+		|   * JSON File, eg: -f="weather.json"
+		""".trimMargin()
+	)
+	exitProcess(0)
+}
+
+private fun checkFile(file: String) {
```

Comment:
In theory this is racy, as file may go away between check and actual open, maybe an open with error check would be better.